mms_msg/sampling/environment/sro.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+from mms_msg.sampling.utils.rng import get_rng
+import paderbox as pb
+
+
+def sample_sro_weights(example, num_nodes, sto_range, sro_range):
+    """
+    Samples the average sapling rate offset and sampling time offset for an example
+    """
+    rng = get_rng(example['example_id'])
+    avg_sro = rng.uniform(*sro_range, size=num_nodes)
+    sto = rng.randint(*sto_range, size=num_nodes)
+    example['avg_sro'] = avg_sro
+    example['sto'] = sto
+    return example
+
+
+def sample_sro(example, sigma=.05, theta=.001, max_sro=400):
+    """
+    Samples a time-varying sampling rate offset (SRO) given an average SRO.
+    Sampling is done via an Ornstein-Uhlenbeck process. For more details, see
+    https://github.com/gfnt/paderwasn
+    """
+    from paderwasn.synchronization.simulation import ornstein_uhlenbeck
+    sig_len = example['num_samples']['observation']
+    sro_seq_len = int(np.ceil((1 + (max_sro * 1e-6)) * sig_len / 2048 + 1))
+    avg_sro = example['avg_sro']
+    sros = []
+    for i, sro in enumerate(avg_sro):
+        seed = pb.utils.random_utils.str_to_seed(example['example_id'] + f'_mic_{i}')
```

Comment:
Use `get_rng_example(example, 'sro_mic', i).randint(2**32)` or use 'example_id' and 'dataset' keys.

We could alternatively extend the `rng` utils and add a `get_seed_from_example` method.